continuous-integration/windows/setup.py

```diff
@@ -6,6 +6,36 @@
 from common import repo_root, run_process, install_pip_dependencies
 
 
+# On Windows the TemporaryDirectory cleanup can sometimes fail.
+# In Python 3.10 and newer, this class has a new argument called
+# `ignore_cleanup_errors` that would help us workaround this issue.
+# However, until we can use Python 3.10, we will need to implement
+# our own workaround.
+class TemporaryDirectoryIgnoringCleanupErrors:
+    def __init__(self):
+        self._tmp_dir = (
+            tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
+        )
+
+    def __enter__(self):
+        return self._tmp_dir.__enter__()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        try:
+            self._tmp_dir.cleanup()
+        except OSError:
+            pass
+        finally:
+            self._tmp_dir = None
+
+    def __fspath__(self):
+        return self.name
```

Comment:
Is this method used anywhere?

Reply:
The AI suggested that I implement this so that it becomes possible to pass it to the constructor of `Path`. Looks like it is a method that is implemented by the `os.PathLike` interface. Didn't try without it though.